mcu/stm32f303.S

```diff
@@ -205,6 +205,22 @@ Reset_Handler:
     strb    r2, [r0, #RCC_AHBRSTR + 0x02]
     strb    r2, [r0, #RCC_AHBENR + 0x02]
     cbz     r4, .L_start_boot
+
+#if ( (defined(DFU_VERIFY_CHECKSUM) && (DFU_VERIFY_CHECKSUM != _DISABLE) ) )
+
+    ldr     r1, =#_APP_START
+    ldr     r2, =__romend
+    sub     r2, r2, r1
+    sub     r2, #8
```

Comment:
As per conversation, the adjustment should use a define from the checksum routine to determine the length, this would allow the checksum to be replaced with SHA or some other more advanced routine.